### PR TITLE
Renamed net to model.

### PR DESCRIPTION
**Pull request by gnovis**

### CODE REVIEW

File: cxflow_tensorflow/hooks/lr_decay_hook.py [hunk 1]
```diff
@@ -34,18 +34,18 @@ class LRDecayHook(AbstractHook):
...
         :param decay_value: the value to modify the learning rate
         :param variable_name: variable name to be modified
         :param decay_type: decay type, one of {'multiply', 'add'}
         """
-        if not isinstance(net, BaseNet):
-            raise TypeError('Invalid net class `{}`. '
-                            'Only to nets derived from `cxflow_tensorflow.BaseNet` are allowed. '.format(type(net)))
+        if not isinstance(model, BaseModel):
+            raise TypeError('Invalid model class `{}`. '
+                            'Only to models derived from `cxflow_tensorflow.BaseModel` are allowed. '.format(type(model)))
```
Comment (FloopCZ) [inline cxflow_tensorflow/hooks/lr_decay_hook.py]: this line seems to be longer than 120 characters

File: cxflow_tensorflow/tests/hooks/lr_decay_hook_test.py [hunk 2]
```diff
@@ -27,43 +27,43 @@ class LRDecayHookTest(TestCase):
...
         repeats = 13
 
-        net = LRNet(dataset=None, log_dir='', inputs=['input', 'target'], outputs=['output'])
-        hook = LRDecayHook(net, decay_value=decay_value)
+        model = LRModel(dataset=None, log_dir='', inputs=['input', 'target'], outputs=['output'])
+        hook = LRDecayHook(model, decay_value=decay_value)
 
         hook.after_epoch()
-        self.assertAlmostEqual(net.graph.get_tensor_by_name('learning_rate:0').eval(session=net.session), 2*decay_value)
+        self.assertAlmostEqual(model.graph.get_tensor_by_name('learning_rate:0').eval(session=model.session), 2*decay_value)
```
Comment (FloopCZ) [inline cxflow_tensorflow/tests/hooks/lr_decay_hook_test.py]: this line seems to be longer than 120 characters

File: cxflow_tensorflow/tests/model_test.py [hunk 3]
```diff
@@ -13,63 +13,63 @@
...
-from cxflow_tensorflow.tests.test_core import CXTestCaseWithDirAndNet
+from cxflow_tensorflow import BaseModel, create_optimizer
+from cxflow_tensorflow.tests.test_core import CXTestCaseWithDirAndModel
 
 
 def create_simple_main_loop(epochs: int, tmpdir: str):
     dataset = SimpleDataset()
-    net = TrainableNet(dataset=dataset, log_dir=tmpdir,  # pylint: disable=redefined-variable-type
+    model = TrainableModel(dataset=dataset, log_dir=tmpdir,  # pylint: disable=redefined-variable-type
                        inputs=['input', 'target'], outputs=['output'])
```
Comment (FloopCZ) [inline cxflow_tensorflow/tests/model_test.py]: unaligned line

File: cxflow_tensorflow/tests/model_test.py [hunk 4]
```diff
@@ -137,198 +137,198 @@ def test_finding_train_op(self):
...
+        saved_var_value = trainable_model.var.eval(session=trainable_model.session)
+        trainable_model.save('1')
+        checkpoint_path = trainable_model.save('2')
 
         tf.reset_default_graph()
 
         # test restoring
-        restored_net = BaseNet(dataset=None, log_dir='', restore_from=self.tmpdir,
+        restored_model = BaseModel(dataset=None, log_dir='', restore_from=self.tmpdir,
                                restore_model_name=checkpoint_path, **trainable_io)
```
Comment (FloopCZ) [inline cxflow_tensorflow/tests/model_test.py]: unaligned line

**Comment by gnovis** [comment]
@FloopCZ Thanks for review, all mentioned bugs should be fixed.

**Pull request merged**